### PR TITLE
Adding workflow to deploy databases to azure

### PR DESCRIPTION
**Pull request by JosephDSchwartz**
The workflow runs the flyway migrate command to run migrations against an Azure Database. I might have to update the set-vars job after https://github.com/im-practices/.github/pull/72 gets merged in.

### CODE REVIEW

File: workflow-templates/im-deploy-az-database.yml
Comment (hpractv) [inline workflow-templates/im-deploy-az-database.yml:40]: Should this be using the new set-environment-variables-by-scope action? 🤓

**Review by hpractv — CHANGES_REQUESTED** [review]
Abbreviated hold until we can get the `set-variables-by-scope` action in place. 😉

File: workflow-templates/im-deploy-az-database.yml
Comment (danielle-casella-adams) [inline workflow-templates/im-deploy-az-database.yml:1]: hehehehe

File: workflow-templates/im-deploy-az-database.yml
Comment (danielle-casella-adams) [inline workflow-templates/im-deploy-az-database.yml:5]: We should re-word for the db specific requirement
```suggestion
# 2 - Make sure the prod runners have access to the database with a 'azurerm_sql_virtual_network_rule' 
```

File: workflow-templates/im-deploy-az-database.yml
Comment (danielle-casella-adams) [inline workflow-templates/im-deploy-az-database.yml:71]: Do we need a corresponding attestor approval block below this?  I'm not sure if teams require QAs to sign off on db deploys or not.
```suggestion
  # Each env has their own stakeholder approval environment.  If no required reviewers are set for
  # that environment, the workflow will continue without requiring anyone to approve the deployment.
  stakeholder-approval:
    needs: [set-vars]
    runs-on: ubuntu-latest # Force this to run on github-hosted runner by using a tag that does not exist on self-hosted runners
    environment: '${{ needs.set-vars.outputs.ENVIRONMENT }} Stakeholder Approval'
    steps:
      - run: echo "The current environment is ${{ needs.set-vars.outputs.ENVIRONMENT }}.  The branch/tag/sha is ${{ env.GITHUB_REF }}."
      - name: Approval Received
        run: echo "Stakeholder approval was received"
  
  # Each env has their own attestor approval environment (meant for QA Attestations).  If no required reviewers are set for
  # that environment, the workflow will continue without requiring anyone to approve the deployment.
  attestor-approval:
    needs: [set-vars]
    runs-on: ubuntu-latest # Force this to run on github-hosted runner by using a tag that does not exist on self-hosted runners
    environment: '${{ needs.set-vars.outputs.ENVIRONMENT }} Attestor Approval'
    steps:
      - run: |
          echo The current environment is ${{ needs.set-vars.outputs.ENVIRONMENT }}.  The branch/tag/sha is ${{ env.GITHUB_REF }}."
          echo "Attestor approval was received or no required reviewers were set for this environment."        
```

File: workflow-templates/im-deploy-az-database.yml
Comment (danielle-casella-adams) [inline workflow-templates/im-deploy-az-database.yml:76]: This potentially needs to be updated with attestor-approval too if you include it above
```suggestion
    needs: [set-vars, stakeholder-approval, attestor-approval]
```

File: workflow-templates/im-deploy-az-database.yml
Comment (danielle-casella-adams) [inline workflow-templates/im-deploy-az-database.yml:69]:
```suggestion
      - run: echo "The current environment is ${{ needs.set-vars.outputs.ENVIRONMENT }}.  The branch/tag/sha is ${{ env.GITHUB_REF }}."
```

File: workflow-templates/im-deploy-az-database.yml
Comment (danielle-casella-adams) [inline workflow-templates/im-deploy-az-database.yml:88]:
```suggestion
      - run: echo "The current environment is ${{ needs.set-vars.outputs.ENVIRONMENT }}.  The branch/tag/sha is ${{ env.GITHUB_REF }}."
```

File: workflow-templates/im-deploy-az-database.yml
Comment (danielle-casella-adams) [inline workflow-templates/im-deploy-az-database.yml:100]: The first step in the job should be printing out the environment/tag


```suggestion
      - run: echo "The current environment is ${{ needs.set-vars.outputs.ENVIRONMENT }}.  The branch/tag/sha is ${{ env.GITHUB_REF }}."

      - name: Checkout Repository
```

File: workflow-templates/im-deploy-az-database.yml
Comment (danielle-casella-adams) [inline workflow-templates/im-deploy-az-database.yml:142]:
```suggestion
      - run: echo "The current environment is ${{ needs.set-vars.outputs.ENVIRONMENT }}.  The branch/tag/sha is ${{ env.GITHUB_REF }}."
```

File: workflow-templates/im-deploy-az-database.yml
Comment (danielle-casella-adams) [inline workflow-templates/im-deploy-az-database.yml:157]:
```suggestion
          ref: ${{ env.GITHUB_REF}}
```

File: workflow-templates/im-deploy-az-database.yml
Comment (danielle-casella-adams) [inline workflow-templates/im-deploy-az-database.yml:23]: If you decide to change this to just tags, the name of this env should be updated to 'RELEASE_TAG' like it is in the other workflows and then all of the places that now reference the GITHUB_TAG would need to be updated.

File: workflow-templates/im-deploy-az-database.yml
Comment (JosephDSchwartz) [reply workflow-templates/im-deploy-az-database.yml:71]: Probably doesn't hurt.

File: workflow-templates/im-deploy-az-database.yml
Comment (JosephDSchwartz) [reply workflow-templates/im-deploy-az-database.yml:15]: I'll add this to the list of updates I'm going to make once the variable substitution stuff gets merged in.

File: workflow-templates/im-deploy-az-database.yml
Comment (danielle-casella-adams) [inline workflow-templates/im-deploy-az-database.yml:115]: I love that it's on linux!

Comment (JosephDSchwartz) [reply workflow-templates/im-deploy-az-database.yml:115]: Me too! I was really happy when I got that working 😃

**Pull request merged**
